src/lib/userStorage/UserProfileStorage.js

```diff
@@ -0,0 +1,494 @@
+    avatar: { defaultPrivacy: 'public' },
+    smallAvatar: { defaultPrivacy: 'public' },
+    walletAddress: { defaultPrivacy: 'public' },
+    username: { defaultPrivacy: 'public' },
+  }
+
+  _lastProfileUpdate: any
+
+  subscribersProfileUpdates = []
+
+  walletAddressIndex = {}
+
+  //unecrypted profile field values
+  profile: { [key: string]: ProfileField } = {}
+
+  constructor(wallet: GoodWallet, profiledb: ProfileDB) {
+    // const seed = Uint8Array.from(Buffer.from(pkeySeed, 'hex'))
+    // this.privateKey = TextileCrypto.PrivateKey.fromRawEd25519Seed(seed)
+    this.wallet = wallet
+    this.profiledb = profiledb
+  }
+
+  /**
+   * read raw profile from database
+   */
+  async init() {
+    const rawProfile = await this.profiledb.getProfile()
+    const decryptedProfile = await this._decryptProfileFields(rawProfile)
+    if (!decryptedProfile.avatar) {
+      decryptedProfile.avatar = { value: null, display: null, privacy: 'public' }
+    }
+    if (!decryptedProfile.smallAvatar) {
+      decryptedProfile.smallAvatar = { value: null, display: null, privacy: 'public' }
+    }
+    this._setLocalProfile(decryptedProfile)
+  }
+
+  /**
+   * helper for setting value of local profile state
+   * @param newValue
+   * @private
+   */
+  _setLocalProfile(newValue) {
+    this.profile = newValue
+    const onProfileUpdate = debounce(
+      doc => {
+        this._lastProfileUpdate = doc
+        over(this.subscribersProfileUpdates)(doc)
+      },
+      500,
+      { leading: false, trailing: true },
+    )
+    onProfileUpdate(newValue)
+  }
+
+  /**
+   * helper for decrypt profile values
+   * @param profile
+   * @returns {Promise<{}>}
+   * @private
+   */
+  async _decryptProfileFields(profile): Promise<any> {
+    const outputProfile = {}
+    await Promise.all(
+      Object.keys(profile).map(
+        async item =>
+          typeof profile[item]?.value === 'string' &&
+          (outputProfile[item] = {
+            ...profile[item],
+            value: await this.profiledb.decryptField(profile[item]?.value),
+          }),
+      ),
+    )
+    return outputProfile
+  }
+
+  /**
+   * encrypt decrypted profile
+   * @param {*} profile
+   */
+  async _encryptProfileFields(profile): Promise<any> {
+    let encryptProfile = {}
+    await Promise.all(
+      Object.keys(profile).map(
+        async field =>
+          typeof profile[field]?.value === 'string' &&
+          (encryptProfile[field] = {
+            ...profile[field],
+            value: await this.profiledb.encryptField(profile[field]?.value),
+          }),
+      ),
+    )
+    return encryptProfile
+  }
+
+  /**
+   * saves a complete profile to the underlying storage
+   * @param {*} profile
+   * @param update
+   */
+  async setProfile(profile, update: boolean = false): Promise<any> {
+    if (profile && !profile.validate) {
+      profile = getUserModel(profile)
+    }
+    const fields = Object.keys(profile).filter(prop => UserStorage.indexableFields[prop])
```

Comment:
should not be dependant on userstorage

Reply:
in the previous userstorage indexable fields were just used to validate certain fields, here you use it to filter what fields to save